state/consulstatedriver.go

```diff
@@ -44,13 +46,22 @@ type ConsulStateDriver struct {
 // Init the driver with a core.Config.
 func (d *ConsulStateDriver) Init(instInfo *core.InstanceInfo) error {
 	var err error
+	var endpoint *url.URL
 
-	if instInfo == nil || !strings.Contains(instInfo.DbURL, "consul://") {
-		return errors.New("invalid consul config")
+	if instInfo == nil || instInfo.DbURL == "" {
+		return errors.New("no Consul config found")
+	}
+	endpoint, err = url.Parse(instInfo.DbURL)
+	if endpoint.Scheme == "consul" {
+		endpoint.Scheme = "http"
+	} else if endpoint.Scheme != "http" && endpoint.Scheme != "https" {
+		return fmt.Errorf("invalid Consul URL scheme %q", endpoint.Scheme)
+	}
+	if err != nil {
```

Comment:
Check error immediately after `url.Parse()` ... if that fails, nothing else matters

Reply:
hum, I recalled i did that ,maybe i missed this one